File: 3252-count-the-number-of-incremovable-subarrays-i/3252-count-the-number-of-incremovable-subarrays-i.go
func getRemovedNums(nums []int, start int, end int) []int {
    copied := make([]int, len(nums))
    copy(copied, nums)
    new_nums := append(copied[:start], copied[end:]...)
    return new_nums
}

func isIncreased(nums []int) bool {
    temp := 0
    for i:=0;i<len(nums);i++{
        if temp >= nums[i] {
            return false
        }
        temp = nums[i]
    }
    return true
}

func incremovableSubarrayCount(nums []int) int {
    var answer int
    for i:=0;i<=len(nums);i++{
        for j:=i+1;j<=len(nums);j++{
            if isIncreased(getRemovedNums(nums, i, j)) {
                answer += 1
            }     
        }
    }
    return answer
}

File: 3252-count-the-number-of-incremovable-subarrays-i/3252-count-the-number-of-incremovable-subarrays-i.py
class Solution:
    def getRemovedNums(self, nums, removed_start, removed_end):
        new_nums = nums[:removed_start] + nums[removed_end:]
        return new_nums
    
    def isIncrease(self, new_nums) -> bool:
        temp = 0
        for num in new_nums:
            if num <= temp: return False
            temp = num
        print(new_nums)
        return True

    def incremovableSubarrayCount(self, nums: List[int]) -> int:
        answer = 0
        for i in range(len(nums)+1):
            for j in range(i+1,len(nums)+1):
               if self.isIncrease(self.getRemovedNums(nums, i, j)):
                answer += 1 
        return answer
